Update Scala 3 version to 3.0.0-M1

tgodzik opened the pull request:
Connected to https://github.com/scalameta/munit/issues/240

Seems we are no longer getting RC versions, so I reverted back to having single stable Dotty version.

tgodzik commented:
I guess this will not be that easy after all :(  I will take a further look tomorrow, but it seems that we are now printing full names with packages. If anyone wants to fix it today, you can just push to this PR with a fix.

gabro commented on build.sbt:
```diff
@@ -11,8 +11,7 @@ def previousVersion = "0.7.0"
 def scala211 = "2.11.12"
-def dottyNext = "0.27.0-RC1"
-def dottyStable = "0.26.0"
+def dottyStable = "3.0.0-M1"
```
how about renaming this to scala3?

tgodzik replied:
Changed it!

tgodzik replied:
And remembered the env variables too in the next push :sweat_smile:

olafurpg reviewed (approved):
Good to get this in

mpilquist commented:
Is this related? https://github.com/lampepfl/dotty/issues/9716

tgodzik commented:
@mpilquist might be, but that issue was also present in the previous versions, so it shouldn't actually change here 🤔

mpilquist commented:
I just noticed this commit in community build -- trying it now: https://github.com/dotty-staging/munit/commit/1e90e9707e02c41049489849dd0e002d48d4b498

mpilquist commented:
Alright figured this one out -- nothing changed in Dotty w.r.t. showing type prefixes. The bug is that we need to change `def isDotty` in `BaseSuite` to account for the version number change.

gabro commented:
Aha, nice catch @mpilquist!

tgodzik commented:
> Alright figured this one out -- nothing changed in Dotty w.r.t. showing type prefixes. The bug is that we need to change `def isDotty` in `BaseSuite` to account for the version number change.

Thanks! Good catch! I updated the PR and also changed the MacroCompat a bit based on the commit you mentioned before.

tgodzik commented:
I will merge and release first thing tomorrow (just in case there is an issue) if no one objects.

mpilquist commented:
Looks good to me. Once munit publishes, we'll be able to start building the various Typelevel libraries. The sooner the better but if you want to wait until tomorrow, that's great. :)

gabro reviewed (approved):
Nice 👏

djspiewak commented:
> Seems we are no longer getting RC versions, so I reverted back to having single stable Dotty version.

I definitely object. It's very very difficult for downstream projects to juggle incompatibilities as the ecosystem updates. Maintaining builds for the two most recent Dotty releases (whenever possible) makes this significantly easier since it allows libraries to take the upgrade in chunks, particularly if and when munit itself makes changes. Publishing for the latest two dotty releases ensures that users are able to split the upgrade pain into two steps: upgrade munit, then upgrade dotty.

Meanwhile, from the perspective of a library maintainer, *not* dropping the previous Dotty release is usually completely painless. Sometimes it has to be done, and particularly with ScalaJS shenanigans I can see merit to dropping 0.27.0-RC1, but I would strongly recommend that publication be for `Mn` and `M(n-1)` from this point forward. We're about to go through the most convoluted and painful cross-publication coordination effort in Scala's history; the least we can do is try to learn from what has made all of the other (otherwise simpler) migrations difficult at times.

tgodzik commented:
> > Seems we are no longer getting RC versions, so I reverted back to having single stable Dotty version.
> 
> I definitely object. It's very very difficult for downstream projects to juggle incompatibilities as the ecosystem updates. Maintaining builds for the two most recent Dotty releases (whenever possible) makes this significantly easier since it allows libraries to take the upgrade in chunks, particularly if and when munit itself makes changes. Publishing for the latest two dotty releases ensures that users are able to split the upgrade pain into two steps: upgrade munit, then upgrade dotty.
> 
> Meanwhile, from the perspective of a library maintainer, _not_ dropping the previous Dotty release is usually completely painless. Sometimes it has to be done, and particularly with ScalaJS shenanigans I can see merit to dropping 0.27.0-RC1, but I would strongly recommend that publication be for `Mn` and `M(n-1)` from this point forward. We're about to go through the most convoluted and painful cross-publication coordination effort in Scala's history; the least we can do is try to learn from what has made all of the other (otherwise simpler) migrations difficult at times.

I can add back `0.27.0-RC1` for sure, might have rushed with dropping it.

olafurpg commented:
@tgodzik feel free to merge and release when this PR is ready!

tgodzik commented:
@djspiewak I added back the `0.27.0-RC1` version. If everything runs ok, I will merge and release it if no one objects?
